### ISSUE TITLE
Constraints involving massless particles

### ISSUE THREAD

**Issue by cwehmeyer** [openmm/openmm#132]
Dear all,

I would like to constrain an entire molecule during an MD simulation to a fix position in space. However, if that molecule is rigid, the default way of setting the atoms' masses to zero yields an exception (see the attached example).

Is it possible to implement a check such that all particles within a constraint group may have zero mass?

Thank you all in advance!

Christoph

Example:

```
import simtk.openmm.app as app
import simtk.openmm as openmm
import simtk.unit as unit

temperature = 300.0 * unit.kelvin
timestep = 2.0 * unit.femtosecond
collision = 1.0 / unit.picosecond
tolerance = 1.0e-8

pdb = app.PDBFile( 'tip3p.pdb' ) # taken from ...python2.7/site-packages/simtk/openmm/app/data/
ff = app.ForceField( 'tip3p.xml' )
system = ff.createSystem( pdb.getTopology(), nonbondedMethod=app.PME, nonbondedCutoff=9*unit.angstrom )
integrator = openmm.LangevinIntegrator( temperature, collision, timestep )
integrator.setConstraintTolerance( tolerance )

### set the masses of the first water molecule to zero
system.setParticleMass( 0, 0.0 )
system.setParticleMass( 1, 0.0 )
system.setParticleMass( 2, 0.0 )

context = openmm.Context( system, integrator )
context.setPositions( pos )

### enforce the constraints
context.applyConstraints( tolerance )
```

Output:

```
Traceback (most recent call last):
  File "constraint-test.py", line 25, in <module>
    context = openmm.Context( system, integrator )
  File "/home/mi/schiffm/local/opt/python/openmm-git/lib/python2.7/site-packages/simtk/openmm/openmm.py", line 3505, in __init__
    this = _openmm.new_Context(*args)
Exception: A constraint cannot involve a massless particle
```

**Comment by stefdoerr** [comment]
I am facing the same issue so I assume there was no update on this. Trying to constrain protein atoms with mass 0 but since I also use `constraints=HBonds` it complains. I ended up using mass 1e10 in the end but it would be nice if the default `mass=0` worked.

**Comment by stefdoerr** [comment]
Actually no. I just noticed that mass `1e10` doesn't work for minimization. The atoms still move.

**Comment by jchodera** [comment]
Minimization should ignore atom masses. 

I think you'll need to set `mass=0` and eliminate constraints where both atoms have `mass=0`, but I agree that it would be a great improvement if OpenMM automatically dropped constraints between paris of zero-mass particles when initializing the context.

**Comment by jchodera** [comment]
@peastman : Maybe tag this as "feature"?

**Comment by stefdoerr** [comment]
Yes. I set them to 0 and removed the HBond constraints to make it work.

**Comment by peastman** [comment]
Sure, that should be an easy thing to do.  Note, though, that it will only help the case where *both* particles are massless.  If you want a constraint where only one of the two particles is massless, that would be more involved.

**Comment by peastman** [comment]
I thought this sounded familiar... it was fixed almost four years ago!  See #155.  Which would explain this issue being closed.  Are you sure this is the problem you're running into?

**Comment by stefdoerr** [comment]
Ok. Since I am constraining atoms by distance to a ligand I assume that I
must be accidentally constraining 1 of the atoms which forms an HBond.
I am going to revise the code anyway for a more intelligent selection of
atoms to constrain in the next few days.
So then the only issue is the error message being slightly misleading (or
not explicative enough).

Thanks a lot for looking into it!
